src/cls/async-listener.ts

```diff
@@ -69,9 +69,15 @@ export class AsyncListenerCLS<Context extends {}> implements CLS<Context> {
     this.getNamespace().set(AsyncListenerCLS.ROOT_CONTEXT_KEY, value);
   }
 
+  clearContext(): void {
+    if (this.getContext() !== this.defaultContext) {
+      this.setContext(this.defaultContext);
+    }
+  }
+
   runWithNewContext<T>(fn: Func<T>): T {
     return this.getNamespace().runAndReturn(() => {
-      this.setContext(this.defaultContext);
+      this.clearContext();
```

Comment:
I am trying to understand why a `clearContext` is necessary here? You didn't change this as part of the PR, but I don't understand the intuition behind this.

Reply:
`clearContext` has been removed. It's unneeded.